src/mono/mono/mini/method-to-ir.c

```diff
@@ -3489,6 +3489,22 @@ method_needs_stack_walk (MonoCompile *cfg, MonoMethod *cmethod)
 		if (!strcmp (cmethod->name, "GetType"))
 			return TRUE;
 	}
+
+#if defined(ENABLE_NETCORE)
```

Comment:
Do we want to do this everywhere or just on wasm?

Reply:
Everywhere really, other platforms need a bit of support as well, just to turn off tail calls.